addons/project/models/project.py

```diff
@@ -593,6 +593,10 @@ def _unlink_except_contains_tasks(self):
         for project in self.with_context(active_test=False):
             if project.tasks:
                 raise UserError(_('You cannot delete a project containing tasks. You can either archive it or first delete all of its tasks.'))
+            external_ids = self.get_external_id()
```

Comment:
should be done outside the for loop to avoid fetching the same data for each project in `self`.